File: backend/src/api/report.ts
import { Hono } from 'hono';
import { renderPlanPdf } from '../skills/report/pdf.js';

export const reportRoute = new Hono();

reportRoute.get('/:id/pdf', async (c) => {
  const id = c.req.param('id');
  const r = await renderPlanPdf(id);
  if (r.ok) {
    return new Response(new Uint8Array(r.bytes), {
      status: 200,
      headers: {
        'content-type': 'application/pdf',
        'content-disposition': `attachment; filename="stackwealth-plan-${id}.pdf"`,
      },
    });
  }
  return new Response(r.html, { status: 200, headers: { 'content-type': 'text/html; charset=utf-8' } });
});

File: backend_py/stackwealth/api/report.py
"""/api/report — PDF render."""
from __future__ import annotations

from fastapi import APIRouter
from fastapi.responses import HTMLResponse, Response

from ..skills.report import render_plan_pdf

router = APIRouter()


@router.get("/{id}/pdf")
async def report_pdf(id: str) -> Response:
    r = await render_plan_pdf(id)
    if r.get("ok"):
        return Response(
            content=r["bytes"],
            media_type="application/pdf",
            headers={
                "content-disposition": f'attachment; filename="stackwealth-plan-{id}.pdf"',
            },
        )
    return HTMLResponse(content=r["html"])
